### DISCUSSION TITLE
connectivity issue with postgresql

### DISCUSSION THREAD

**Discussion by sharukhkhan234** [horilla-opensource/horilla#234] [Q&A]
HIi,
I have connected the Horilla project with PostgreSQL, and added the necessary changes in the settings.py file as needed for the connectivity with Postgres, An issue that I have is I am unable to log-in because my database is empty. The server also runs successfully, but when try to log-in in to the browser,getting the notification that your password or username does not match. Further created a superuser and checked with that superuser username and password, and it generated a new issue. I requested to please help me regarding this issue, I have tried a lot, but still stuck

**Comment by horilla-opensource** [comment]
Can you please share the screenshot of the changes you have made in the settings.py file ?

**Reply by sharukhkhan234** [reply]
![postgres](https://github.com/horilla-opensource/horilla/assets/157000111/b66c52bb-2a94-4a21-a13c-cde10d5c794f)

![setting](https://github.com/horilla-opensource/horilla/assets/157000111/a1ac821e-d184-40f6-bf35-4951508a529b)

**Reply by horilla-opensource** [reply]
Did you create any user using the _createhorillauser_ command ?
If not, please share the screenshot of the login page.

With Regards,
**Team Horilla**

**Reply by sharukhkhan234** [reply]
Actually, i created a Horilla user with the createhorillauser command when installing the Horilla with the default database, but as shifted to the new Postgres database, did not create (createhorillauser) again, should I create a Horilla user again?

**Reply by sharukhkhan234** [reply]
![login](https://github.com/horilla-opensource/horilla/assets/157000111/1ddfaff1-025d-4e17-8a46-4cd2718a5484)

the server is running correctly but I tried with a username and password which I have added in the settings.py file, but I am unable to log-in it.

**Reply by horilla-opensource** [reply]
Hi @sharukhkhan234 ,
If you are using the latest code base, you'll get an option in the login page for _initialize database._ I think its not the latest code base.
Also the username and password that you provide inside the _settings.py_ file is for accessing the database (postgres) and not for accessing Horilla. To access Horilla, you'll have to create a record in the users model, for that you'll have to make use of the _createhorillauser_ command.

Since you are starting with a new database, there won't be any data inside that database, so you'll have to start by creating a user first.
```
python manage.py createhorillauser
```
The above command will help you create a user.

As I told above, if you are using the latest code, when there is no super admin user created in the database, you'll get an option in the login page (Initialize Database) with which you can create the first admin user from the frontend itself along with company details, departments etc.

With Regards,
**Team Horilla**

**Reply by horilla-opensource** [reply]
<img width="1470" alt="Screenshot 2024-07-05 at 10 33 40 AM" src="https://github.com/horilla-opensource/horilla/assets/131998600/ef25982b-0dd7-4a92-b0b9-c9ad21a59dec">
<img width="1470" alt="Screenshot 2024-07-05 at 10 34 52 AM" src="https://github.com/horilla-opensource/horilla/assets/131998600/41e21071-fbf3-4949-b15c-eeb8718f63d9">

**Reply by sharukhkhan234** [reply]
a month ago I installed this on Linux, should I reinstall it for the latest code or createhorillauser? or if I reinstall it, can I install this software with the Postgres database or I should convert it after being installed?

**Reply by horilla-opensource** [reply]
No need to re install, just pull the latest code and use the same configuration as of now.
Also if you reinstall, you just need to change the database settings in the settings.py file to add the postgres configuration in it.

**Reply by sharukhkhan234** [reply]
how do I pull the latest code? 
did you mean to download the latest code and paste it on the environment repository? or something else

**Reply by horilla-opensource** [reply]
How did you start the project initially?
Did you downloaded the zip code or cloned the repo or pull from the repo ?

**Reply by sharukhkhan234** [reply]
git clone https://github.com/horilla-opensource/horilla.git

cloned by following the above command and followed all the commands as provided.

![horilla_installation](https://github.com/horilla-opensource/horilla/assets/157000111/c80a61a4-2230-49ab-9305-c2a38befd4a1)

**Reply by horilla-opensource** [reply]
Okay.
To pull the latest code, go to the project directory and run the following command.
```
git pull https://github.com/horilla-opensource/horilla.git master
```

**Reply by sharukhkhan234** [reply]
and what to do with rather commands, like -r requirements.txt?

**Reply by horilla-opensource** [reply]
Yes, run the requirements.txt file just to confirm the latest packages are installed.
After that run the makemigration and migrate command.

**Reply by sharukhkhan234** [reply]
in which directory should I pull the (git pull https://github.com/horilla-opensource/horilla.git master) this?

![directories](https://github.com/horilla-opensource/horilla/assets/157000111/8eee7722-cede-4e92-ab48-de07c9f9adea)

**Reply by horilla-opensource** [reply]
Your current directory, horilla(directory which includes all the apps and files)

**Reply by sharukhkhan234** [reply]
logged-in  successfully.
Thank you!

Regards,
